crates/runtime/src/instance/allocator/pooling/unix.rs

```diff
@@ -0,0 +1,64 @@
+use anyhow::{bail, Context, Result};
+
+fn decommit(addr: *mut u8, len: usize, protect: bool) -> Result<()> {
+    if len == 0 {
+        return Ok(());
+    }
+
+    if unsafe {
+        libc::mmap(
```

Comment:
As this is a fixed mapping at the same address, it effectively discards the existing pages and remaps them to the CoW zero page (because of being both private and `PROT_READ`|`PROT_WRITE`).